### ISSUE TITLE
Video is smooth and rarely drops frame in handheld, but constantly and rhythmically drops frames docked

### ISSUE THREAD

**Issue by techniponi** [RSDuck/vba-next-switch#22]
RCM payload version: `Hekate - CTCaer mod v2.3`
hbmenu version: `v2.0.0`
SDFiles version: `v2.0` (from [here](https://github.com/tumGER/SDFilesSwitch/releases/tag/v2.0))
VBA-Next-switch version: `0.6`

Tested on both a Samsung HDTV and an Elgato HD60 Pro. Docked mode runs quite smooth, and frame drops are present but infrequent. In docked, however, several frames are dropped per second and the entire experience is very stuttery and unpleasant. I can record footage if you would like me to.

**Comment by techniponi** [comment]
Because I'm bored and have nothing better to do, here's a recording from OBS with my capture card.
Switch is running at 1080p with full HDMI color range, if it matters.
[2018-06-10 02-08-13.zip](https://github.com/RSDuck/vba-next-switch/files/2087608/2018-06-10.02-08-13.zip)

**Comment by RSDuck** [comment]
Thanks for reporting! I already knew about this, but since I rarely play in docked mode, I haven't thought much about it.

I don't think there's a straightforward fix to this, I might have to experiment a bit.

**Comment by techniponi** [comment]
I figured the best way to bugtest something is "try to play it normally and complain whenever something goes wrong"

**Comment by RSDuck** [comment]
yeah, that's always a good method of testing. I'm currently playing through Pokemon Emerald for the first time(I already played Saphire and Omega Ruby, so not exactly for the first time) and now I really want to implement netplay😄

**Comment by techniponi** [comment]
Emerald is the best rendition of Hoenn in my opinion. ORAS are very inconsistent and hit-and-miss with their changes, and while I still play and re-play ORAS mostly due to the improved battle mechanics, I think Emerald is still the best way to experience the region and story.

Netplay would be excellent for trading and battling, but the problem is the way the GBA expects latency - the entire game will freeze and wait for the data to be transferred Every. Single. Frame.

It expects absolutely 0.0ms ping, so anything higher than that tends to be a borderline unplayable mess.

If you can somehow make it work better though, I'd genuinely be amazed.

**Comment by techniponi** [comment]
Although, now that I think about it, perhaps LOCAL splitscreen (as in, running several instances of a game on the same system) could be feasible.

For games like Kirby and the Amazing Mirror and TLoZ: Four Swords, there is definitely appeal in splitscreen gaming. The only hard thing to implement is keeping track of save files I suppose (perhaps the splitscreen mode would require the user to manually name/select a save profile?)

Since it's all local, this circumvents the ping problem, and would definitely be a super nice thing to have for games like the ones I mentioned.

**Comment by Keristero** [comment]
Hi, I also noticed this issue, Hope it can be resolved without too much difficulty

**Comment by RSDuck** [comment]
I'm currently away from home and don't have a TV nearby. The newest version, which was released today might have fixed the issue but probably hasn't.

@Keristero or someone else, can you please tell me whether you're using the newest version of VBA-next and also if the same issue appears in other homebrew apps(which tbh I haven't tested, I use the docked mode only rarely).
